pallets/order-book/src/lib.rs

```diff
@@ -334,8 +338,28 @@ pub mod pallet {
 			Ok(())
 		}
 
-		///  Cancel an existing order that had been created by calling account.
+		/// Update an existing order
 		#[pallet::call_index(1)]
+		#[pallet::weight(T::Weights::create_order())]
+		pub fn user_update_order(
+			origin: OriginFor<T>,
+			order_id: T::OrderIdNonce,
+			buy_amount: T::Balance,
+			price: T::SellRatio,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+			let order = <Orders<T>>::get(order_id)?;
+			if account_id == order.placing_account {
+				<Self as TokenSwaps<T::AccountId>>::update_order(
+					account_id, order_id, buy_amount, price, buy_amount,
+				)
+			} else {
+				Err(DispatchError::from(Error::<T>::Unauthorised))
+			}
```

Comment:
```suggestion
      ensure!(account_id == order.placing_account, Error::<T>::Unauthorised);
			
			<Self as TokenSwaps<T::AccountId>>::update_order(
					account_id, order_id, buy_amount, price, buy_amount,
			)
```
`nit` but more idiomatic